packages/components/src/ActionList/utils/useActionListSortManager.tsx

```diff
@@ -0,0 +1,117 @@
+/*
+
+ MIT License
+
+ Copyright (c) 2020 Looker Data Sciences, Inc.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+
+import React, { useState } from 'react'
+import {
+  ActionList,
+  ActionListColumns,
+  ActionListItem,
+  ActionListItemColumn,
+  ActionListItemAction,
+} from '..'
+
+const stringComparator = (stringA: string, stringB: string) => {
+  const upperCasedStringA = stringA.toUpperCase()
+  const upperCasedStringB = stringB.toUpperCase()
+
+  if (upperCasedStringA < upperCasedStringB) return -1
+  if (upperCasedStringA > upperCasedStringB) return 1
+  return 0
+}
+
+type ActionListData = Record<string, string | number>
+
+export const useActionListSortManager = (
+  actionListData: ActionListData[],
+  actionListColumns: ActionListColumns
+) => {
+  const [data, setData] = useState(actionListData)
+  const [columns, setColumns] = useState(actionListColumns)
+
+  const MyActions = () => (
+    <>
+      <ActionListItemAction onClick={() => alert(`You performed an action!`)}>
+        Some Action
+      </ActionListItemAction>
+    </>
```

Comment:
Doesn't need a fragment around:

```suggestion
      <ActionListItemAction onClick={() => alert(`You performed an action!`)}>
        Some Action
      </ActionListItemAction>
```

As a follow-up to this PR we might need to think about how this would actually be useful. For someone to use this it seems this will need to be something that a developer can specify as a function – something like a prop that `useActionListSortManager` accepts:

```
actions: (item: ActionListData) => ReactNode
```

And then is implemented as:

```
const actionGenerator = (item: ActionListData) => (
  <ActionListItemAction onClick={callFunction(item.columns[0]}}>
        Send Item: "{item detail here]}" to Someone... 
      </ActionListItemAction>
)
```

Reply:
Resolved in 011c45c!